### PR TITLE
Export etl metrics to prometheus

### LINKED ISSUE #1248: Export ETL metrics to Prometheus
We should export all the data except `etl_sources` provided by `EtlService::getInfo()` to Prometheus.

### PR DESCRIPTION
**Pull request by kuznetsss**
Fixes #1248.

### CODE REVIEW

**Comment by kuznetsss** [comment]
I also removed global `PrometheusService` instance from tests and added `WithPrometheus` where needed. Now if a test uses Prometheus counters (even somewhere under the hood) it will be failing by assertion. It is much better because forgetting about the global `PrometheusService` is leading to confusing situation when you think you just created a new variable but it has already not default value because the value is from some previous test with Prometheus counters.

File: src/etl/SystemState.hpp [hunk 1]
```diff
@@ -33,9 +37,19 @@ struct SystemState {
      * In strict read-only mode, the process will never attempt to become the ETL writer, and will only publish ledgers
      * as they are written to the database.
      */
-    bool isReadOnly = false;
+    util::prometheus::Bool isReadOnly = PrometheusService::boolMetric(
+        "etl_read_only",
```
Comment (cindyyan317) [inline src/etl/SystemState.hpp]: better change it to read_only to distinguish the etl which is not writing at the moment

**Review by cindyyan317 — DISMISSED** [review]
LGTM , just saw some name not fit

File: src/util/prometheus/Bool.hpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
```
Comment (godexsoft) [inline src/util/prometheus/Bool.hpp]: 2024 probably

File: unittests/util/prometheus/BoolTests.cpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
```
Comment (godexsoft) [inline unittests/util/prometheus/BoolTests.cpp]: 2024 also

**Review by godexsoft — DISMISSED** [review]
Noticed few small things only - code looks good to me 👍

**Review by godexsoft — APPROVED** [review]

**Pull request merged**